### ISSUE TITLE
Encrypted Streams API is not OOP-friendly

### ISSUE THREAD

**Issue by andreiled** [paragonie/sodium-plus#38]
## Problem Statement
Current _Encrypted Streams_ API falls short of the claim that _Sodium-Plus is the libsodium API that JavaScript developers deserve_ made in the [Sodium-Plus: A Positive Cryptography Experience for JavaScript Developers](https://dev.to/paragonie/sodium-plus-a-positive-cryptography-experience-for-javascript-developers-2p08) blog post.
Specifically:
1. It looks like a classic C API rather than an object-oriented API: `push` and `pull` method require to explicitly pass raw reference to the `state` object returned by the corresponding `init` method.
This not only makes it harder to understand how to use these APIs (even with the [documentation](https://github.com/paragonie/sodium-plus/blob/master/docs/SodiumPlus/encrypted-streams.md)), but also makes the code look heavier.
1. It exposes an internal `state` object akin to a raw pointer in C/C++ increasing the risk of misusing it.
1. It is too easy for a developer to mess up and call a wrong flavor of the `push`/`pull` method, especially if/when additional algorithms aside from _XChaCha20 Poly1305_ will be added in future (I know that the goal is to avoid having an unnecessarily wide selection of algorithms to prevent developers from accidentally choosing a bad one, but having at least a couple of algorithms available would certainly be beneficial in some situations).

## Solution
I suggest to:
1. Update `crypto_secretstream_xchacha20poly1305_init_push` to return encryptor object with:
    * `header` read-only property
    * `push(message, ad = '', tag = 0)` method
    * `rekey` method
1. Update `crypto_secretstream_xchacha20poly1305_init_pull` method to return a new decryptor object with:
    * `pull(ciphertext, ad = '', tag = 0)` method

This change is **not** backward-compatible, but I still consider it acceptable since:
* Based on the `crypto_secretstream_xchacha20poly1305_init_*` method names alone, they look perfect for the job.
In other words, I believe that coming up with new weird names to house the new interface would create a confusion for the initiated when choosing the right method to use.
* Considering, how _hidden_ this particular API is, I believe it is safe to assume that there aren't to many usages of it.
